File: Source/Service/appService/com.maxiaohua.genealogy.main.biz/src/com/maxiaohua/genealogy/main/biz/a03/dto/NoticeCommentDTO.java
/*
 * Genealogy
 * Xiao Hua Ma personal studio 2015.
 */
package com.maxiaohua.genealogy.main.biz.a03.dto;

/**
 * NoticeCommentDTO DTO
 *
 * History
 * REV.         Updated Date           Updater              Infomation
 * -------      ---------------        ----------------     ------------------
 * 1.0          2015/09/06             TOOL                Create
*
 */
@com.maxiaohua.genealogy.fw.core.type.AliasKanJi("通知评论")
public class NoticeCommentDTO implements java.io.Serializable, Comparable<NoticeCommentDTO> {
	/**
	* System ID
	*/
	private static final long serialVersionUID = 1L;

	private String userFace = null;

	public String getUserFace() {
		return this.userFace;
	}

	public void setUserFace(
			String userFace) {
		this.userFace = userFace;
	}

	private String userName = null;

	public String getUserName() {
		return this.userName;
	}

	public void setUserName(
			String userName) {
		this.userName = userName;
	}

	private java.sql.Date date = null;

	public java.sql.Date getDate() {
		return this.date;
	}

	public void setDate(
			java.sql.Date date) {
		this.date = date;
	}

	private java.sql.Time time = null;

	public java.sql.Time getTime() {
		return this.time;
	}

	public void setTime(
			java.sql.Time time) {
		this.time = time;
	}

	private String comment = null;

	public String getComment() {
		return this.comment;
	}

	public void setComment(
			String comment) {
		this.comment = comment;
	}

  	/* (non-Javadoc)
     * @see java.lang.Comparable#compareTo(java.lang.Object)
     */
    @Override
    public int compareTo(
    		NoticeCommentDTO compareToDTO) {
        return this.equals(compareToDTO) ? 0 : -1;
    }
}

File: Source/iOS/Genealogy/Genealogy/Model/A03/DTO/NoticeCommentDTO.m
//
//  NoticeCommentDTO.m
//  Genealogy
//
//  Created by xiao huama on 15/09/10.
//  Xiao Hua Ma personal studio 2015.
//

#import "NoticeCommentDTO.h"

@implementation NoticeCommentDTO

- (void)loadDataWithJsonData:(id)jsonData
{
    if (![jsonData isKindOfClass:[NSDictionary class]]) {
        return;
    }
    if (!jsonData) {
        return;
    }
    if (![jsonData[@"userFace"] isKindOfClass:[NSNull class]]) {
        self.userFace = jsonData[@"userFace"];
    }
    if (![jsonData[@"userName"] isKindOfClass:[NSNull class]]) {
        self.userName = jsonData[@"userName"];
    }
    if (![jsonData[@"date"] isKindOfClass:[NSNull class]]) {
        self.date = jsonData[@"date"];
    }
    if (![jsonData[@"time"] isKindOfClass:[NSNull class]]) {
        self.time = jsonData[@"time"];
    }
    if (![jsonData[@"comment"] isKindOfClass:[NSNull class]]) {
        self.comment = jsonData[@"comment"];
    }
}

-(NSMutableDictionary *)buildRequestData
{
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    if (self.userFace) {
        dic[@"userFace"] = self.userFace;
    }
    if (self.userName) {
        dic[@"userName"] = self.userName;
    }
    if (self.date) {
        dic[@"date"] = self.date;
    }
    if (self.time) {
        dic[@"time"] = self.time;
    }
    if (self.comment) {
        dic[@"comment"] = self.comment;
    }
    return dic;
}

@end
